recipes-tests/trenchboot-tests/trenchboot-tests/cukinia.conf

```diff
@@ -0,0 +1,53 @@
+export CUKINIA_ROOT=/etc/cukinia
+
+shopt -s nullglob
+
+check_binary() {
+    files=("/boot/$1"*.bin)
+    if [ ${#files[@]} -eq 0 ]; then
+        return 1
+    fi
+    for file in "${files[@]}"; do
+        if [ -f "$file" ]; then
+            grep "/$1" /boot/grub/grub.cfg || return 1
+        else
+            return 1
+        fi
+    done
+}
+
+acms=(
+    ADL_SINIT
+    BDW_SINIT
+    CFL_SINIT
+    CMLSTGP_SINIT
+    CML_RKL_S_SINIT
+    CML_S_SINIT
+    RKLS_SINIT
+    SKL_KBL_AML_SINIT
+    SNB_IVB_SINIT
+    TGL_SINIT
+)
+
+cukinia_log "$(_colorize yellow "--- GRUB ---")"
+cukinia_test -f /boot/grub/grub.cfg
+cukinia_test -f /boot/EFI/BOOT/grub.cfg
+as "Check if /boot/grub/grub.cfg and /boot/EFI/BOOT/grub.cfg are the same" \
+    cukinia_cmd cmp /boot/grub/grub.cfg /boot/EFI/BOOT/grub.cfg
+as "grub.cfg contains slaunch" cukinia_cmd grep 'slaunch\s*$' /boot/grub/grub.cfg
+
+cukinia_log "$(_colorize yellow "--- ACM & SKL ---")"
+for acm in "${acms[@]}"; do
+    as "/boot/acm/$acm*.bin exists and is used in grub.cfg" cukinia_cmd check_binary "acm/$acm"
+done
+as "/boot/skl.bin exists and is used in grub.cfg" cukinia_cmd check_binary "skl"
+
+cukinia_log "$(_colorize yellow "--- packages ---")"
+as "Check whether the tpm2_tools package can be started" cukinia_cmd tpm2_eventlog -v
```

Comment:
Added in [c3db2e7](https://github.com/3mdeb/meta-trenchboot/pull/33/commits/c3db2e718f3956976b28938929acf2d8db019440) in kernel part.
I fail the test immediately if TPM version is 1.2 otherwise `tpm2_pcrread` takes long time before failing.
Will TrenchBoot be used only on machines with TPM 2.0? If not then I can skip this test instead.

Reply:
No, we want both TPM 1.2 and 2.0 to work.

However, now that I think about it, we don't need `tpm2_tools` at all. We can read PCRs from sysfs in both cases, and this is all we need for now. It wasn't the case on older kernels, which only exported PCRs for TPM 1.2.

Reply:
That's what I was doing before deleting it as @macpijan suggested.
What should I do, bring back those tests or delete this one as PCR tests will be implemented in RF?

Reply:
> What should I do, bring back those tests or delete this one as PCR tests will be implemented in RF?

We need to decide on something. My rule of thumb would be to not implement in cukinia anything that does not work in qemu, requires any scenario more complex than simply starting a qemu, or requires more than a few lines to implement - so it is much more complex than simply executing a command and checking exit code.

Reply:
@macpijan Ok, removed in [12c72a9](https://github.com/3mdeb/meta-trenchboot/pull/33/commits/12c72a937484adb40cd14fa60d12eb1392e72e9f).
Test would work but only if run in OSFV QEMU (run with `scripts/ci/qemu-run.sh` with few changes to boot external image) so not too simple to set up at least first time.

Reply:
I still think that having confirmed access to PCR values in sysfs is important, something like these two lines did: https://github.com/3mdeb/meta-trenchboot/commit/c154d49d2a9143b60d3d5822d38569af73024ca4#diff-e34f41a7042cb64c2d2d8ae3198c4261e6828713c4a0ea884f8b6a8cd7f5ef90L55-L56. Is it really worth to remove tests for crucial functionality _expected from meta-trenchboot image_ (not the platform it runs on) just so we can skip few arguments passed to QEMU?

Reply:
@krystian-hebel Added in [5412283](https://github.com/3mdeb/meta-trenchboot/pull/33/commits/54122834f9ea9d9fb9c9c1fef0072f439fa44b89)

```
[PASS] Checking kernel config "TCG_TPM" set to "y"
[PASS] Checking kernel config "TCG_TIS" set to "y"
[PASS] Checking kernel config "TCG_TIS_CORE" set to "y"
```